src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -97,7 +74,47 @@ public Task<ExportJobOutcome> GetExportJobByHashAsync(string hash, CancellationT
 
         public Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobRecord, WeakETag eTag, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            // TODO: Implement this method.
+            return Task.FromResult(new ExportJobOutcome(jobRecord, eTag));
+        }
+
+        public async Task<IReadOnlyCollection<ExportJobOutcome>> AcquireExportJobsAsync(ushort maximumNumberOfConcurrentJobsAllowed, TimeSpan jobHeartbeatTimeoutThreshold, CancellationToken cancellationToken)
+        {
+            // We will consider a job to be stale if its timestamp is smaller than or equal to this.
+            DateTimeOffset expirationTime = Clock.UtcNow - jobHeartbeatTimeoutThreshold;
+
+            using (SqlConnectionWrapper sqlConnectionWrapper = _sqlConnectionWrapperFactory.ObtainSqlConnectionWrapper(true))
+            using (SqlCommand sqlCommand = sqlConnectionWrapper.CreateSqlCommand())
+            {
+                V1.AcquireExportJobs.PopulateCommand(
+                    sqlCommand,
+                    expirationTime,
+                    maximumNumberOfConcurrentJobsAllowed);
+
+                var acquiredJobs = new List<ExportJobOutcome>();
+
+                using (SqlDataReader sqlDataReader = await sqlCommand.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                {
+                    while (await sqlDataReader.ReadAsync(cancellationToken))
+                    {
+                        (string rawJobRecord, byte[] rowVersionAsBytes) = sqlDataReader.ReadRow(V1.ExportJob.RawJobRecord, V1.ExportJob.JobVersion);
+
+                        var exportJobRecord = JsonConvert.DeserializeObject<ExportJobRecord>(rawJobRecord);
+
+                        if (BitConverter.IsLittleEndian)
+                        {
+                            Array.Reverse(rowVersionAsBytes);
+                        }
+
+                        const int startIndex = 0;
+                        var rowVersionAsDecimalString = BitConverter.ToInt64(rowVersionAsBytes, startIndex).ToString();
```

Comment:
nit: why not just use named parameter here? e.g., BitConverter.ToInt64(rowVersionAsBytes, startIndex: 0)?

Reply:
Thanks, fixed in 9f3d322.